[Telink] Add Telink TLX Module Board & Update compatible builds to docker version 150

wuhaoTelink opened the pull request:
#### Summary

- Update compatible builds to docker version 150 (from 140)
    - Keep [NRF] image on version 140 (require additional changes after https://github.com/project-chip/connectedhomeip/pull/39702)
    - Keep [TI CC32XX] image on version 115
- Style fixes in `local.py` required to pass lint after the Docker/linter update
  → Previous failed run (before fixes): https://github.com/project-chip/connectedhomeip/actions/runs/16289933090/job/45997276746
- Add new Telink module board and updated Telink build targets

#### Testing

- Update builds to docker version 150 verified by GitHub CI
- Lint passes with updated style https://github.com/project-chip/connectedhomeip/actions/runs/16290098548/job/45997815472?pr=39536
- Telink board builds successfully with the new configuration
    Manual test:
    - west build -b tl7218x_ml7g
    - west build -b tl7218x_ml7m 
    - west build -b tl3218x_ml3m

wuhaoTelink commented on examples/lighting-app/telink/src/tflm/audio/app_codec.c:
```diff
@@ -59,7 +59,7 @@ void app_codec_init()
     if (CONFIG_SOC_RISCV_TELINK_TL721X)
     {
-        audio_set_stream0_dmic_pin(GPIO_FC_PA2, GPIO_FC_PA3, GPIO_FC_PA4);
+        audio_set_stream0_dmic_pin(GPIO_FC_PB4, GPIO_FC_PA2, GPIO_NONE_PIN);
```
Adapt DMIC for TLX Module Board.

wuhaoTelink commented on examples/lighting-app/telink/src/tflm/audio/app_config.h:
```diff
@@ -14,8 +14,8 @@ extern "C" {
 #define SYNC_SAMPLE 16 * (DATA_TRANS_INTERVAL_MS)
-#define CATEGORY_YES_TRIGGER_THRESHOLD 9000 // 90%
-#define CATEGORY_NO_TRIGGER_THRESHOLD 7500  // 75%
+#define CATEGORY_YES_TRIGGER_THRESHOLD 8000 // 80%
+#define CATEGORY_NO_TRIGGER_THRESHOLD 7000  // 70%
```
The Voice recognition performance on the module board is improved when CATEGORY_YES_TRIGGER_THRESHOLD  is set to 8000 and CATEGORY_NO_TRIGGER_THRESHOLD  is set to 7000. You can modify it to suit the module board you have.

andy31415 reviewed (changes requested):
@wuhaoTelink please fill out the PR summary. it seems all sections are left blank (e.g. how was this tested?)

andy31415 commented on examples/lighting-app/telink/src/tflm/audio/app_config.h:
```diff
@@ -14,8 +14,8 @@ extern "C" {
 #define SYNC_SAMPLE 16 * (DATA_TRANS_INTERVAL_MS)
-#define CATEGORY_YES_TRIGGER_THRESHOLD 9000 // 90%
-#define CATEGORY_NO_TRIGGER_THRESHOLD 7500  // 75%
+#define CATEGORY_YES_TRIGGER_THRESHOLD 8000 // 80%
```
These adjustment seem independent of the adding of a new board. Please separate changes in distinct PRs.

wuhaoTelink replied:
I have separated them, and now the PR is just to add Telink TLX module board. Please help check it, thanks!

wuhaoTelink committed:
[Telink] Add Telink TLX Module Board

- add tl7218x modual board ml7g
- add tl7218x modual board ml7m
- add tl3218x modual board ml3m

Signed-off-by: Hao Wu <hao.wu@telink-semi.com>

wuhaoTelink committed:
Merge remote-tracking branch 'origin/master' into add_tlx_modual_board

s07641069 committed:
Merge branch 'master' into add_tlx_modual_board

s07641069 committed:
[Telink] Update hash for test

s07641069 committed:
Merge branch 'master' into add_tlx_modual_board

s07641069 committed:
[Telink] Update compatible builds to docker version 146

s07641069 committed:
[Telink] update image versions

s07641069 committed:
[Telink] Use available image

s07641069 committed:
Merge branch 'master' into add_tlx_modual_board

s07641069 committed:
[Telink] Update compatible builds to docker version 150

s07641069 committed:
[Telink] fix after rebase

restyled-commits committed:
Restyled by prettier-markdown

s07641069 committed:
Merge branch 'master' into add_tlx_modual_board

s07641069 committed:
fix code-lints

andy31415 commented on scripts/tools/file_size_from_nm.py:
```diff
@@ -487,12 +487,12 @@ def build_treemap(
             else:
                 # The "full name" is generally quite long, so shorten it...
-                data["name_with_size"][idx] = f"{data["short_name"][idx]}: {total_size}"
+                data["name_with_size"][idx] = f"{data['short_name'][idx]}: {total_size}"
```
These changes could be a separate PR (probably with the explanation that "" that is nested is hard to follow or maybe does not work in some python versions or some other explanation...

s07641069 replied:
These style changes are actually required to pass lint after updating the Docker image version.
Without them, the lint job fails — here's the failed run:
https://github.com/project-chip/connectedhomeip/actions/runs/16289933090/job/45997276746

So unless we revert the Docker update, these fixes are needed to keep CI passed.
I updated the PR description to clarify this.

s07641069 committed:
[Telink] add build targets
